smartsim/_core/control/controller.py

```diff
@@ -591,3 +599,113 @@ def reload_saved_db(self, checkpoint_file):
+        devices = self._enumerate_devices(db_model)
+        try:
+            client = Client(address=address, cluster=cluster)
+        except RedisConnectionError as error:
+            logger.error("Could not connect to orchestrator")
+            raise error
+
+        for device in devices:
+            try:
+                if db_model.is_file:
+                    client.set_model_from_file(
+                        name=db_model.name,
+                        model_file=str(db_model.file),
+                        backend=db_model.backend,
+                        device=device,
+                        batch_size=db_model.batch_size,
+                        min_batch_size=db_model.min_batch_size,
+                        tag=db_model.tag,
+                        inputs=db_model.inputs,
+                        outputs=db_model.outputs
+                    )
+                else:
+                    client.set_model(
+                        name=db_model.name,
+                        model=db_model.model,
+                        backend=db_model.backend,
+                        device=device,
+                        batch_size=db_model.batch_size,
+                        min_batch_size=db_model.min_batch_size,
+                        tag=db_model.tag,
+                        inputs=db_model.inputs,
+                        outputs=db_model.outputs
+                    )
+            except  RedisReplyError as error:
+                logger.error("Error while setting model on orchestrator.")
+                raise error
+
+
+    def _set_script(self, db_script, address, cluster=False):   
+        devices = self._enumerate_devices(db_script)
+        try:
+            client = Client(address=address, cluster=cluster)
+        except RedisConnectionError as error:
+            logger.error("Could not connect to orchestrator")
+            raise error
+
+        for device in devices:
+            try:
+                if db_script.is_file:
+                    client.set_script_from_file(
+                        name=db_script.name,
+                        file=str(db_script.file),
+                        device=device
+                    )
+                else:
+                    if isinstance(db_script.script, str):
+                        client.set_script(
+                            name=db_script.name,
+                            script=db_script.script,
+                            device=device
+                        )
+                    else:
+                        client.set_function(
+                            name=db_script.name,
+                            function=db_script.script,
+                            device=device
+                        )
+        
+            except  RedisReplyError as error:
+                logger.error("Error while setting model on orchestrator.")
+                raise error
+
+
+    def _set_dbobjects(self, manifest):
```

Comment:
this will stay in the ``Controller`` (probably obviously)

Reply:
Yep, it stayed there with some mods